R/cv_varsel.R

```diff
@@ -823,24 +830,18 @@ loo_varsel <- function(refmodel, method, nterms_max, ndraws,
       if (length(mssgs_warns_capt) > 0) {
         warning(mssgs_warns_capt)
       }
+
       if (importance_sampling_nm == "psis") {
         pareto_k_eval <- loo::pareto_k_values(sub_psisloo)
         warn_pareto(
-          n07 = sum(pareto_k_eval > 0.7),
-          n05 = sum(0.7 >= pareto_k_eval & pareto_k_eval > 0.5),
-          warn_txt_start = paste0(
-            "In the recalculation of the reference model's PSIS-LOO CV ",
-            "weights for the performance evaluation (based on clustered or ",
-            "thinned posterior draws), "
-          ),
-          warn_txt_mid_common = paste0(
-            " (out of ", nloo, ") Pareto k-values are "
-          ),
-          warn_txt_end = paste0(
-            ". Watch out for warnings thrown by the original-draws Pareto ",
-            "smoothing to see whether it makes sense to increase the number ",
-            "of draws (resulting from the clustering or thinning for the ",
-            "performance evaluation). Alternatively, K-fold CV can be used."
+          n07 = sum(pareto_k_eval > .ps_khat_threshold(dim(psisloo)[1])), n = n,
+          khat_threshold = .ps_khat_threshold(dim(sub_psisloo)[1]),
+          warn_txt = paste0(
+            "Some (%d / % d) Pareto k's for the reference model's PSIS-LOO weights for ",
+            ifelse(clust_used_eval,
+                   paste0(nclusters_pred, " clustered "),
+                   paste0(ndraws_pred, " posterior ")),
+            "draws are > %s."
```

Comment:
This now generates warnings such as
```
Warning: Some (7 /  100) Pareto k's for the reference model's PSIS-LOO weights for 30 posterior draws are > 0.32.
```
in case of thinning. I'm not sure if this will be easy to understand for users (because this lacks the reference to thinning now and also because the double occurrence of "for" makes it slightly hard to read), but I'm fine with keeping this if you prefer so.

Reply:
fixed

Reply:
thanks